### ISSUE TITLE
Upgrade Java Version to JDK 11

### ISSUE THREAD

**Issue by RohanNagar** [RohanNagar/thunder#112]
Now that Dropwizard 1.3.0 is out and officially supports JDK 1.9, we can migrate Thunder to use Java 9. This would be a big step and we should probably move to using modules.

Java 10 is also now out, and moving to Java 9 would mean the Java 10 upgrade is a lot easier. In Java 10, we can do cool stuff like use `var` to declare local variables.

Marking this as low priority, but will keep this on the radar.

https://dzone.com/articles/a-practical-guide-to-java-9-migration
https://github.com/dropwizard/dropwizard/issues/2318

**Comment by RohanNagar** [comment]
Checklist:

- [ ] Compile with Java 9
- [ ] Change Travis and GitLab to build with JDK 9
- [ ] Change `Dockerfile` to use JRE 9
- [ ] Update Bootstrap script to install JDK 9 instead of 8

**Comment by RohanNagar** [comment]
PR is open for this - but blocking on other improvements to testing infrastructure so that we can be sure nothing is breaking.

**Comment by RohanNagar** [comment]
Right now this is blocked on a new version of Dropwizard that has better support for Java 9+.

Also, Java 11 will be released soon, so that should be the new target.
